iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/util/IidmXmlUtil.java

```diff
@@ -57,92 +66,196 @@ default void run() {
         }
     }
 
+    private static String message(String elementName, ErrorMessage type, IidmXmlVersion version, IidmXmlVersion contextVersion, String reason) {
+        return elementName + " is " + type.message + " for IIDM-XML version " + contextVersion.toString(".") + ". " + reason + version.toString(".");
+    }
+
     private static PowsyblException createException(String rootElementName, String elementName, ErrorMessage type, IidmXmlVersion version, IidmXmlVersion contextVersion, String reason) {
         return createException(rootElementName + "." + elementName, type, version, contextVersion, reason);
     }
 
     private static PowsyblException createException(String elementName, ErrorMessage type, IidmXmlVersion version, IidmXmlVersion contextVersion, String reason) {
-        return new PowsyblException(elementName + " is " + type.message + " for IIDM-XML version " + contextVersion.toString(".") + ". " + reason + version.toString("."));
+        return new PowsyblException(message(elementName, type, version, contextVersion, reason));
+    }
+
+    private static void createExceptionOrLogError(String rootElementName, String elementName, ErrorMessage type, IidmXmlVersion refVersion, String reason, NetworkXmlWriterContext context) {
+        createExceptionOrLogError(rootElementName + "." + elementName, type, refVersion, reason, context);
+    }
+
+    private static void createExceptionOrLogError(String elementName, ErrorMessage type, IidmXmlVersion refVersion, String reason, NetworkXmlWriterContext context) {
+        if (context.getOptions().getIidmVersionIncompatibilityBehavior() == ExportOptions.IidmVersionIncompatibilityBehavior.THROW_EXCEPTION) {
+            throw createException(elementName, type, refVersion, context.getVersion(), reason);
+        } else if (context.getOptions().getIidmVersionIncompatibilityBehavior() == ExportOptions.IidmVersionIncompatibilityBehavior.LOG_ERROR) {
+            Supplier<String> message = () -> message(elementName, type, refVersion, context.getVersion(), reason);
+            LOGGER.error(message.get());
```

Comment:
There is no good reason to create a supplier if you always call the `get()` method. I would replace it by a simple string and check if ERROR level is enabled